SAMLPortal/Controllers/AppController.cs

```diff
@@ -0,0 +1,219 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using SAMLPortal.Misc;
+using SAMLPortal.Models;
+using Microsoft.EntityFrameworkCore;
+using ITfoxtec.Identity.Saml2.Schemas.Metadata;
+using SAMLPortal.Misc;
+
+namespace SAMLPortal.Controllers
+{
+	[Route("App")]
+	[Authorize(Roles = UserRoles.Administrator)]
+	public class AppController : Controller
+	{
+		// GET: App
+		public ActionResult Index()
+		{
+			SAMLPortalContext context = new SAMLPortalContext();
+			return View(context.App.ToList());
+		}
+
+		// GET: App/Details/5
+		[Route("Details/{id}")]
+		public ActionResult Details(int id)
+		{
+			SAMLPortalContext context = new SAMLPortalContext();
+			App app = context.App.Find(id);
+			return View(app);
+		}
+
+		// GET: App/Create
+		[HttpGet]
+		[Route("Create")]
+		public ActionResult Create()
+		{
+			App emptyApp = new App();
+			return View(emptyApp);
+		}
+
+		// Post: App/Create
+		[HttpPost]
+		[ValidateAntiForgeryToken]
+		[Route("Create")]
+		public async Task<IActionResult> Create(App appli)
+		{
+			try
+			{
+				SAMLPortalContext context = new SAMLPortalContext();
+				if (ModelState.IsValid)
+				{
+					if (appli.MetadataURL != null)
+					{
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Either remove or fill this block of code.](https://app.codacy.com/manual/zegorax/SAMLPortal/pullRequest?prid=5434886)